### ISSUE TITLE
async_wrapper does not filter garbage from module output

### ISSUE THREAD

**Issue by samvarankashyap** [ansible/ansible#17035]
##### ISSUE TYPE

- Bug Report
##### COMPONENT NAME

module : os_server and async_status
##### ANSIBLE VERSION

```
ansible 2.1.0.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides
```
##### SUMMARY

Tried using os_server module in using async , its failing to handle , output of the job .
##### STEPS TO REPRODUCE

Create a playbook to boot a single instance on openstack async 

```

---
- name: test the os_server module on async
  hosts: localhost
  connection: local
  gather_facts: false
  tasks:
    - name: "provision os_server resources"
      os_server:
        state: "present"
        auth:
          auth_url: "http://localhost:5000/v2.0/"
          username: "openstackusername"
          password: "openstackpassword"
          project_name: "openstackprojectname"
        name: "helloinstance"
        image: "rhel-6.5_jeos"
        key_name: "test_keypair"
        api_timeout: 99999
        flavor: "m1.small"
        network: "testnetwork"
      async: 1000
      poll: 0
      register: yum_sleeper
    - name: 'check on fire and forget task'
      async_status:
        jid: "{{ yum_sleeper.ansible_job_id }}"
      register: job_result
      until: job_result.finished
      retries: 30

```

##### EXPECTED RESULTS

Expected , job output with openstack server details . 
##### ACTUAL RESULTS

https://gist.github.com/samvarankashyap/645de866d564eee9e2fe0fcb6c02c77e

command:

```
ansible-playbook -vvvvvv pluck_os.yml 
```

Actual output:
https://gist.github.com/samvarankashyap/645de866d564eee9e2fe0fcb6c02c77e

```

Using /etc/ansible/ansible.cfg as config file
 [WARNING]: provided hosts list is empty, only localhost is available

Loaded callback default of type stdout, v2.0

PLAYBOOK: pluck_os.yml *********************************************************
1 plays in pluck_os.yml

PLAY [test the os_server module on async] **************************************

TASK [provision/deprovision os_server resources by looping on count] ***********
task path: /root/linch-pin/pluck_os.yml:7
<127.0.0.1> ESTABLISH LOCAL CONNECTION FOR USER: root
<127.0.0.1> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo $HOME/.ansible/tmp/ansible-tmp-1470853932.03-182394085106578 `" && echo ansible-tmp-1470853932.03-182394085106578="` echo $HOME/.ansible/tmp/ansible-tmp-1470853932.03-182394085106578 `" ) && sleep 0'                                                                               
<127.0.0.1> PUT /tmp/tmpF3aQ4d TO /root/.ansible/tmp/ansible-tmp-1470853932.03-182394085106578/os_server
<127.0.0.1> PUT /tmp/tmpQqJAgL TO /root/.ansible/tmp/ansible-tmp-1470853932.03-182394085106578/async_wrapper
<127.0.0.1> EXEC /bin/sh -c 'chmod -R u+x /root/.ansible/tmp/ansible-tmp-1470853932.03-182394085106578/ && sleep 0'
<127.0.0.1> EXEC /bin/sh -c 'LANG=en_US.UTF-8 LC_ALL=en_US.UTF-8 LC_MESSAGES=en_US.UTF-8 /root/.ansible/tmp/ansible-tmp-1470853932.03-182394085106578/async_wrapper 501588893815 1000 /root/.ansible/tmp/ansible-tmp-1470853932.03-182394085106578/os_server && sleep 0'                                                                          
<127.0.0.1> EXEC /bin/sh -c 'rm -f -r /root/.ansible/tmp/ansible-tmp-1470853932.03-182394085106578/ > /dev/null 2>&1 && sleep 0'
ok: [localhost] => {"ansible_job_id": "501588893815.445", "changed": false, "results_file": "/root/.ansible_async/501588893815.445", "started": 1}

TASK [check on fire and forget task] *******************************************
task path: /root/linch-pin/pluck_os.yml:24
<127.0.0.1> ESTABLISH LOCAL CONNECTION FOR USER: root
<127.0.0.1> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo $HOME/.ansible/tmp/ansible-tmp-1470853933.23-277604438370373 `" && echo ansible-tmp-1470853933.23-277604438370373="` echo $HOME/.ansible/tmp/ansible-tmp-1470853933.23-277604438370373 `" ) && sleep 0'                                                                               
<127.0.0.1> PUT /tmp/tmpIkSVL9 TO /root/.ansible/tmp/ansible-tmp-1470853933.23-277604438370373/async_status
<127.0.0.1> EXEC /bin/sh -c 'LANG=en_US.UTF-8 LC_ALL=en_US.UTF-8 LC_MESSAGES=en_US.UTF-8 /usr/bin/python /root/.ansible/tmp/ansible-tmp-1470853933.23-277604438370373/async_status; rm -rf "/root/.ansible/tmp/ansible-tmp-1470853933.23-277604438370373/" > /dev/null 2>&1 && sleep 0'                                                           
FAILED - RETRYING: TASK: check on fire and forget task (29 retries left).Result was: {"ansible_job_id": "501588893815.445", "attempts": 1, "changed": false, "finished": 0, "invocation": {"module_args": {"jid": "501588893815.445", "mode": "status"}, "module_name": "async_status"}, "results_file": "/root/.ansible_async/501588893815.445", "retries": 30, "started": 1}
<127.0.0.1> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo $HOME/.ansible/tmp/ansible-tmp-1470853938.32-242588126297954 `" && echo ansible-tmp-1470853938.32-242588126297954="` echo $HOME/.ansible/tmp/ansible-tmp-1470853938.32-242588126297954 `" ) && sleep 0'                                                                               
<127.0.0.1> PUT /tmp/tmp4jd2U7 TO /root/.ansible/tmp/ansible-tmp-1470853938.32-242588126297954/async_status
<127.0.0.1> EXEC /bin/sh -c 'LANG=en_US.UTF-8 LC_ALL=en_US.UTF-8 LC_MESSAGES=en_US.UTF-8 /usr/bin/python /root/.ansible/tmp/ansible-tmp-1470853938.32-242588126297954/async_status; rm -rf "/root/.ansible/tmp/ansible-tmp-1470853938.32-242588126297954/" > /dev/null 2>&1 && sleep 0'                                                           
FAILED - RETRYING: TASK: check on fire and forget task (28 retries left).Result was: {"ansible_job_id": "501588893815.445", "attempts": 2, "changed": false, "finished": 0, "invocation": {"module_args": {"jid": "501588893815.445", "mode": "status"}, "module_name": "async_status"}, "results_file": "/root/.ansible_async/501588893815.445", "retries": 30, "started": 1}
<127.0.0.1> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo $HOME/.ansible/tmp/ansible-tmp-1470853943.42-153837274674906 `" && echo ansible-tmp-1470853943.42-153837274674906="` echo $HOME/.ansible/tmp/ansible-tmp-1470853943.42-153837274674906 `" ) && sleep 0'                                                                               
<127.0.0.1> PUT /tmp/tmpchWAZ5 TO /root/.ansible/tmp/ansible-tmp-1470853943.42-153837274674906/async_status
<127.0.0.1> EXEC /bin/sh -c 'LANG=en_US.UTF-8 LC_ALL=en_US.UTF-8 LC_MESSAGES=en_US.UTF-8 /usr/bin/python /root/.ansible/tmp/ansible-tmp-1470853943.42-153837274674906/async_status; rm -rf "/root/.ansible/tmp/ansible-tmp-1470853943.42-153837274674906/" > /dev/null 2>&1 && sleep 0'                                                           
FAILED - RETRYING: TASK: check on fire and forget task (27 retries left).Result was: {"ansible_job_id": "501588893815.445", "attempts": 3, "changed": false, "finished": 0, "invocation": {"module_args": {"jid": "501588893815.445", "mode": "status"}, "module_name": "async_status"}, "results_file": "/root/.ansible_async/501588893815.445", "retries": 30, "started": 1}
<127.0.0.1> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo $HOME/.ansible/tmp/ansible-tmp-1470853948.51-150710275020038 `" && echo ansible-tmp-1470853948.51-150710275020038="` echo $HOME/.ansible/tmp/ansible-tmp-1470853948.51-150710275020038 `" ) && sleep 0'                                                                               
<127.0.0.1> PUT /tmp/tmpTFyDJs TO /root/.ansible/tmp/ansible-tmp-1470853948.51-150710275020038/async_status
<127.0.0.1> EXEC /bin/sh -c 'LANG=en_US.UTF-8 LC_ALL=en_US.UTF-8 LC_MESSAGES=en_US.UTF-8 /usr/bin/python /root/.ansible/tmp/ansible-tmp-1470853948.51-150710275020038/async_status; rm -rf "/root/.ansible/tmp/ansible-tmp-1470853948.51-150710275020038/" > /dev/null 2>&1 && sleep 0'                                                           
FAILED - RETRYING: TASK: check on fire and forget task (26 retries left).Result was: {"ansible_job_id": "501588893815.445", "attempts": 4, "changed": false, "finished": 0, "invocation": {"module_args": {"jid": "501588893815.445", "mode": "status"}, "module_name": "async_status"}, "results_file": "/root/.ansible_async/501588893815.445", "retries": 30, "started": 1}
<127.0.0.1> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo $HOME/.ansible/tmp/ansible-tmp-1470853953.6-191645118737367 `" && echo ansible-tmp-1470853953.6-191645118737367="` echo $HOME/.ansible/tmp/ansible-tmp-1470853953.6-191645118737367 `" ) && sleep 0'                                                                                  
<127.0.0.1> PUT /tmp/tmpFrI9Bt TO /root/.ansible/tmp/ansible-tmp-1470853953.6-191645118737367/async_status
<127.0.0.1> EXEC /bin/sh -c 'LANG=en_US.UTF-8 LC_ALL=en_US.UTF-8 LC_MESSAGES=en_US.UTF-8 /usr/bin/python /root/.ansible/tmp/ansible-tmp-1470853953.6-191645118737367/async_status; rm -rf "/root/.ansible/tmp/ansible-tmp-1470853953.6-191645118737367/" > /dev/null 2>&1 && sleep 0'                                                             
fatal: [localhost]: FAILED! => {"ansible_job_id": "501588893815.445", "changed": false, "failed": true, "finished": 1, "invocation": {"module_args": {"jid": "501588893815.445", "mode": "status"}, "module_name": "async_status"}, "msg": "Could not parse job output: No handlers could be found for logger \"keystoneauth.identity.base\"\n\n{\"invocation\": {\"module_args\": {\"auth_type\": null, \"availability_zone\": null, \"image\": \"rhel-6.5_jeos\", \"image_exclude\": \"(deprecated)\", \"flavor_include\": null, \"meta\": null, \"flavor\": \"m1.small\", \"cloud\": null, \"scheduler_hints\": null, \"boot_from_volume\": false, \"userdata\": null, \"network\": \"VALUE_SPECIFIED_IN_NO_LOG_PARAMETER\", \"nics\": [], \"floating_ips\": null, \"flavor_ram\": null, \"volume_size\": false, \"state\": \"present\", \"auto_ip\": true, \"security_groups\": [\"default\"], \"config_drive\": false, \"volumes\": [], \"key_name\": \"ci-factory\", \"api_timeout\": 99999, \"auth\": {\"username\": \"VALUE_SPECIFIED_IN_NO_LOG_PARAMETER\", \"project_name\": \"VALUE_SPECIFIED_IN_NO_LOG_PARAMETER\", \"password\": \"VALUE_SPECIFIED_IN_NO_LOG_PARAMETER\", \"auth_url\": \"VALUE_SPECIFIED_IN_NO_LOG_PARAMETER\"}, \"endpoint_type\": \"public\", \"boot_volume\": null, \"key\": null, \"cacert\": null, \"wait\": true, \"name\": \"helloinstance\", \"region_name\": null, \"timeout\": 180, \"cert\": null, \"terminate_volume\": false, \"verify\": true, \"floating_ip_pools\": null}}, \"openstack\": {\"OS-EXT-STS:task_state\": null, \"addresses\": {\"e2e-openstack\": [{\"OS-EXT-IPS-MAC:mac_addr\": \"fa:16:3e:11:38:38\", \"version\": 4, \"addr\": \"172.16.100.97\", \"OS-EXT-IPS:type\": \"fixed\"}, {\"OS-EXT-IPS-MAC:mac_addr\": \"fa:16:3e:11:38:38\", \"version\": 4, \"addr\": \"10.8.183.233\", \"OS-EXT-IPS:type\": \"floating\"}]}, \"image\": {\"id\": \"3bcfd17c-6bf0-4134-ae7f-80bded8b46fd\", \"name\": \"rhel-6.5_jeos\"}, \"OS-EXT-STS:vm_state\": \"active\", \"OS-SRV-USG:launched_at\": \"2016-08-10T18:32:23.000000\", \"NAME_ATTR\": \"name\", \"flavor\": {\"id\": \"2\", \"name\": \"m1.small\"}, \"az\": \"nova\", \"id\": \"b66ba857-5740-4c13-9e70-191883398af8\", \"cloud\": \"defaults\", \"user_id\": \"9c770dbddda444799e627004fee26e0a\", \"OS-DCF:diskConfig\": \"MANUAL\", \"networks\": {\"e2e-openstack\": [\"172.16.100.97\", \"10.8.183.233\"]}, \"accessIPv4\": \"10.8.183.233\", \"accessIPv6\": \"\", \"security_groups\": [{\"id\": \"df1a797b-009c-4685-a7c9-43863c36d653\", \"name\": \"default\", \"security_group_rules\": [{\"direction\": \"ingress\", \"protocol\": null, \"remote_ip_prefix\": null, \"port_range_max\": null, \"security_group_id\": \"df1a797b-009c-4685-a7c9-43863c36d653\", \"port_range_min\": null, \"ethertype\": \"IPv4\", \"id\": \"ade9fcb9-14c1-4975-a04d-6007f80005c1\"}, {\"direction\": \"ingress\", \"protocol\": null, \"remote_ip_prefix\": null, \"port_range_max\": null, \"security_group_id\": \"df1a797b-009c-4685-a7c9-43863c36d653\", \"port_range_min\": null, \"ethertype\": \"IPv4\", \"id\": \"d03e4bae-24b6-415a-a30c-ee0d060f566f\"}], \"description\": \"Default security group\"}], \"key_name\": \"ci-factory\", \"progress\": 0, \"OS-EXT-STS:power_state\": 1, \"OS-EXT-AZ:availability_zone\": \"nova\", \"metadata\": {}, \"status\": \"ACTIVE\", \"updated\": \"2016-08-10T18:32:23Z\", \"hostId\": \"be958de354ca4b72bb0a02694148f8d7f2d5ba965cb49e864fe63d37\", \"HUMAN_ID\": true, \"OS-SRV-USG:terminated_at\": null, \"public_v4\": \"10.8.183.233\", \"public_v6\": \"\", \"private_v4\": \"172.16.100.97\", \"interface_ip\": \"10.8.183.233\", \"name\": \"helloinstance\", \"created\": \"2016-08-10T18:32:16Z\", \"tenant_id\": \"f1dda47890754241a3e111f9b7394707\", \"region\": \"\", \"adminPass\": \"tX4SZ3gV85Sb\", \"os-extended-volumes:volumes_attached\": [], \"volumes\": [], \"config_drive\": \"\", \"human_id\": \"helloinstance\"}, \"changed\": true, \"id\": \"b66ba857-5740-4c13-9e70-191883398af8\", \"server\": {\"OS-EXT-STS:task_state\": null, \"addresses\": {\"e2e-openstack\": [{\"OS-EXT-IPS-MAC:mac_addr\": \"fa:16:3e:11:38:38\", \"version\": 4, \"addr\": \"172.16.100.97\", \"OS-EXT-IPS:type\": \"fixed\"}, {\"OS-EXT-IPS-MAC:mac_addr\": \"fa:16:3e:11:38:38\", \"version\": 4, \"addr\": \"10.8.183.233\", \"OS-EXT-IPS:type\": \"floating\"}]}, \"image\": {\"id\": \"3bcfd17c-6bf0-4134-ae7f-80bded8b46fd\", \"name\": \"rhel-6.5_jeos\"}, \"OS-EXT-STS:vm_state\": \"active\", \"OS-SRV-USG:launched_at\": \"2016-08-10T18:32:23.000000\", \"NAME_ATTR\": \"name\", \"flavor\": {\"id\": \"2\", \"name\": \"m1.small\"}, \"az\": \"nova\", \"id\": \"b66ba857-5740-4c13-9e70-191883398af8\", \"cloud\": \"defaults\", \"user_id\": \"9c770dbddda444799e627004fee26e0a\", \"OS-DCF:diskConfig\": \"MANUAL\", \"networks\": {\"e2e-openstack\": [\"172.16.100.97\", \"10.8.183.233\"]}, \"accessIPv4\": \"10.8.183.233\", \"accessIPv6\": \"\", \"security_groups\": [{\"id\": \"df1a797b-009c-4685-a7c9-43863c36d653\", \"name\": \"default\", \"security_group_rules\": [{\"direction\": \"ingress\", \"protocol\": null, \"remote_ip_prefix\": null, \"port_range_max\": null, \"security_group_id\": \"df1a797b-009c-4685-a7c9-43863c36d653\", \"port_range_min\": null, \"ethertype\": \"IPv4\", \"id\": \"ade9fcb9-14c1-4975-a04d-6007f80005c1\"}, {\"direction\": \"ingress\", \"protocol\": null, \"remote_ip_prefix\": null, \"port_range_max\": null, \"security_group_id\": \"df1a797b-009c-4685-a7c9-43863c36d653\", \"port_range_min\": null, \"ethertype\": \"IPv4\", \"id\": \"d03e4bae-24b6-415a-a30c-ee0d060f566f\"}], \"description\": \"Default security group\"}], \"key_name\": \"ci-factory\", \"progress\": 0, \"OS-EXT-STS:power_state\": 1, \"OS-EXT-AZ:availability_zone\": \"nova\", \"metadata\": {}, \"status\": \"ACTIVE\", \"updated\": \"2016-08-10T18:32:23Z\", \"hostId\": \"be958de354ca4b72bb0a02694148f8d7f2d5ba965cb49e864fe63d37\", \"HUMAN_ID\": true, \"OS-SRV-USG:terminated_at\": null, \"public_v4\": \"10.8.183.233\", \"public_v6\": \"\", \"private_v4\": \"172.16.100.97\", \"interface_ip\": \"10.8.183.233\", \"name\": \"helloinstance\", \"created\": \"2016-08-10T18:32:16Z\", \"tenant_id\": \"f1dda47890754241a3e111f9b7394707\", \"region\": \"\", \"adminPass\": \"tX4SZ3gV85Sb\", \"os-extended-volumes:volumes_attached\": [], \"volumes\": [], \"config_drive\": \"\", \"human_id\": \"helloinstance\"}}\n{\"msg\": \"Traceback (most recent call last):\\n  File \\\"/root/.ansible/tmp/ansible-tmp-1470853932.03-182394085106578/async_wrapper\\\", line 89, in _run_module\\n  File \\\"/usr/lib64/python2.7/json/__init__.py\\\", line 339, in loads\\n    return _default_decoder.decode(s)\\n  File \\\"/usr/lib64/python2.7/json/decoder.py\\\", line 364, in decode\\n    obj, end = self.raw_decode(s, idx=_w(s, 0).end())\\n  File \\\"/usr/lib64/python2.7/json/decoder.py\\\", line 382, in raw_decode\\n    raise ValueError(\\\"No JSON object could be decoded\\\")\\nValueError: No JSON object could be decoded\\n\", \"failed\": 1, \"cmd\": \"/root/.ansible/tmp/ansible-tmp-1470853932.03-182394085106578/os_server\", \"data\": \"No handlers could be found for logger \\\"keystoneauth.identity.base\\\"\\n\\n{\\\"invocation\\\": {\\\"module_args\\\": {\\\"auth_type\\\": null, \\\"availability_zone\\\": null, \\\"image\\\": \\\"rhel-6.5_jeos\\\", \\\"image_exclude\\\": \\\"(deprecated)\\\", \\\"flavor_include\\\": null, \\\"meta\\\": null, \\\"flavor\\\": \\\"m1.small\\\", \\\"cloud\\\": null, \\\"scheduler_hints\\\": null, \\\"boot_from_volume\\\": false, \\\"userdata\\\": null, \\\"network\\\": \\\"VALUE_SPECIFIED_IN_NO_LOG_PARAMETER\\\", \\\"nics\\\": [], \\\"floating_ips\\\": null, \\\"flavor_ram\\\": null, \\\"volume_size\\\": false, \\\"state\\\": \\\"present\\\", \\\"auto_ip\\\": true, \\\"security_groups\\\": [\\\"default\\\"], \\\"config_drive\\\": false, \\\"volumes\\\": [], \\\"key_name\\\": \\\"ci-factory\\\", \\\"api_timeout\\\": 99999, \\\"auth\\\": {\\\"username\\\": \\\"VALUE_SPECIFIED_IN_NO_LOG_PARAMETER\\\", \\\"project_name\\\": \\\"VALUE_SPECIFIED_IN_NO_LOG_PARAMETER\\\", \\\"password\\\": \\\"VALUE_SPECIFIED_IN_NO_LOG_PARAMETER\\\", \\\"auth_url\\\": \\\"VALUE_SPECIFIED_IN_NO_LOG_PARAMETER\\\"}, \\\"endpoint_type\\\": \\\"public\\\", \\\"boot_volume\\\": null, \\\"key\\\": null, \\\"cacert\\\": null, \\\"wait\\\": true, \\\"name\\\": \\\"helloinstance\\\", \\\"region_name\\\": null, \\\"timeout\\\": 180, \\\"cert\\\": null, \\\"terminate_volume\\\": false, \\\"verify\\\": true, \\\"floating_ip_pools\\\": null}}, \\\"openstack\\\": {\\\"OS-EXT-STS:task_state\\\": null, \\\"addresses\\\": {\\\"e2e-openstack\\\": [{\\\"OS-EXT-IPS-MAC:mac_addr\\\": \\\"fa:16:3e:11:38:38\\\", \\\"version\\\": 4, \\\"addr\\\": \\\"172.16.100.97\\\", \\\"OS-EXT-IPS:type\\\": \\\"fixed\\\"}, {\\\"OS-EXT-IPS-MAC:mac_addr\\\": \\\"fa:16:3e:11:38:38\\\", \\\"version\\\": 4, \\\"addr\\\": \\\"10.8.183.233\\\", \\\"OS-EXT-IPS:type\\\": \\\"floating\\\"}]}, \\\"image\\\": {\\\"id\\\": \\\"3bcfd17c-6bf0-4134-ae7f-80bded8b46fd\\\", \\\"name\\\": \\\"rhel-6.5_jeos\\\"}, \\\"OS-EXT-STS:vm_state\\\": \\\"active\\\", \\\"OS-SRV-USG:launched_at\\\": \\\"2016-08-10T18:32:23.000000\\\", \\\"NAME_ATTR\\\": \\\"name\\\", \\\"flavor\\\": {\\\"id\\\": \\\"2\\\", \\\"name\\\": \\\"m1.small\\\"}, \\\"az\\\": \\\"nova\\\", \\\"id\\\": \\\"b66ba857-5740-4c13-9e70-191883398af8\\\", \\\"cloud\\\": \\\"defaults\\\", \\\"user_id\\\": \\\"9c770dbddda444799e627004fee26e0a\\\", \\\"OS-DCF:diskConfig\\\": \\\"MANUAL\\\", \\\"networks\\\": {\\\"e2e-openstack\\\": [\\\"172.16.100.97\\\", \\\"10.8.183.233\\\"]}, \\\"accessIPv4\\\": \\\"10.8.183.233\\\", \\\"accessIPv6\\\": \\\"\\\", \\\"security_groups\\\": [{\\\"id\\\": \\\"df1a797b-009c-4685-a7c9-43863c36d653\\\", \\\"name\\\": \\\"default\\\", \\\"security_group_rules\\\": [{\\\"direction\\\": \\\"ingress\\\", \\\"protocol\\\": null, \\\"remote_ip_prefix\\\": null, \\\"port_range_max\\\": null, \\\"security_group_id\\\": \\\"df1a797b-009c-4685-a7c9-43863c36d653\\\", \\\"port_range_min\\\": null, \\\"ethertype\\\": \\\"IPv4\\\", \\\"id\\\": \\\"ade9fcb9-14c1-4975-a04d-6007f80005c1\\\"}, {\\\"direction\\\": \\\"ingress\\\", \\\"protocol\\\": null, \\\"remote_ip_prefix\\\": null, \\\"port_range_max\\\": null, \\\"security_group_id\\\": \\\"df1a797b-009c-4685-a7c9-43863c36d653\\\", \\\"port_range_min\\\": null, \\\"ethertype\\\": \\\"IPv4\\\", \\\"id\\\": \\\"d03e4bae-24b6-415a-a30c-ee0d060f566f\\\"}], \\\"description\\\": \\\"Default security group\\\"}], \\\"key_name\\\": \\\"ci-factory\\\", \\\"progress\\\": 0, \\\"OS-EXT-STS:power_state\\\": 1, \\\"OS-EXT-AZ:availability_zone\\\": \\\"nova\\\", \\\"metadata\\\": {}, \\\"status\\\": \\\"ACTIVE\\\", \\\"updated\\\": \\\"2016-08-10T18:32:23Z\\\", \\\"hostId\\\": \\\"be958de354ca4b72bb0a02694148f8d7f2d5ba965cb49e864fe63d37\\\", \\\"HUMAN_ID\\\": true, \\\"OS-SRV-USG:terminated_at\\\": null, \\\"public_v4\\\": \\\"10.8.183.233\\\", \\\"public_v6\\\": \\\"\\\", \\\"private_v4\\\": \\\"172.16.100.97\\\", \\\"interface_ip\\\": \\\"10.8.183.233\\\", \\\"name\\\": \\\"helloinstance\\\", \\\"created\\\": \\\"2016-08-10T18:32:16Z\\\", \\\"tenant_id\\\": \\\"f1dda47890754241a3e111f9b7394707\\\", \\\"region\\\": \\\"\\\", \\\"adminPass\\\": \\\"tX4SZ3gV85Sb\\\", \\\"os-extended-volumes:volumes_attached\\\": [], \\\"volumes\\\": [], \\\"config_drive\\\": \\\"\\\", \\\"human_id\\\": \\\"helloinstance\\\"}, \\\"changed\\\": true, \\\"id\\\": \\\"b66ba857-5740-4c13-9e70-191883398af8\\\", \\\"server\\\": {\\\"OS-EXT-STS:task_state\\\": null, \\\"addresses\\\": {\\\"e2e-openstack\\\": [{\\\"OS-EXT-IPS-MAC:mac_addr\\\": \\\"fa:16:3e:11:38:38\\\", \\\"version\\\": 4, \\\"addr\\\": \\\"172.16.100.97\\\", \\\"OS-EXT-IPS:type\\\": \\\"fixed\\\"}, {\\\"OS-EXT-IPS-MAC:mac_addr\\\": \\\"fa:16:3e:11:38:38\\\", \\\"version\\\": 4, \\\"addr\\\": \\\"10.8.183.233\\\", \\\"OS-EXT-IPS:type\\\": \\\"floating\\\"}]}, \\\"image\\\": {\\\"id\\\": \\\"3bcfd17c-6bf0-4134-ae7f-80bded8b46fd\\\", \\\"name\\\": \\\"rhel-6.5_jeos\\\"}, \\\"OS-EXT-STS:vm_state\\\": \\\"active\\\", \\\"OS-SRV-USG:launched_at\\\": \\\"2016-08-10T18:32:23.000000\\\", \\\"NAME_ATTR\\\": \\\"name\\\", \\\"flavor\\\": {\\\"id\\\": \\\"2\\\", \\\"name\\\": \\\"m1.small\\\"}, \\\"az\\\": \\\"nova\\\", \\\"id\\\": \\\"b66ba857-5740-4c13-9e70-191883398af8\\\", \\\"cloud\\\": \\\"defaults\\\", \\\"user_id\\\": \\\"9c770dbddda444799e627004fee26e0a\\\", \\\"OS-DCF:diskConfig\\\": \\\"MANUAL\\\", \\\"networks\\\": {\\\"e2e-openstack\\\": [\\\"172.16.100.97\\\", \\\"10.8.183.233\\\"]}, \\\"accessIPv4\\\": \\\"10.8.183.233\\\", \\\"accessIPv6\\\": \\\"\\\", \\\"security_groups\\\": [{\\\"id\\\": \\\"df1a797b-009c-4685-a7c9-43863c36d653\\\", \\\"name\\\": \\\"default\\\", \\\"security_group_rules\\\": [{\\\"direction\\\": \\\"ingress\\\", \\\"protocol\\\": null, \\\"remote_ip_prefix\\\": null, \\\"port_range_max\\\": null, \\\"security_group_id\\\": \\\"df1a797b-009c-4685-a7c9-43863c36d653\\\", \\\"port_range_min\\\": null, \\\"ethertype\\\": \\\"IPv4\\\", \\\"id\\\": \\\"ade9fcb9-14c1-4975-a04d-6007f80005c1\\\"}, {\\\"direction\\\": \\\"ingress\\\", \\\"protocol\\\": null, \\\"remote_ip_prefix\\\": null, \\\"port_range_max\\\": null, \\\"security_group_id\\\": \\\"df1a797b-009c-4685-a7c9-43863c36d653\\\", \\\"port_range_min\\\": null, \\\"ethertype\\\": \\\"IPv4\\\", \\\"id\\\": \\\"d03e4bae-24b6-415a-a30c-ee0d060f566f\\\"}], \\\"description\\\": \\\"Default security group\\\"}], \\\"key_name\\\": \\\"ci-factory\\\", \\\"progress\\\": 0, \\\"OS-EXT-STS:power_state\\\": 1, \\\"OS-EXT-AZ:availability_zone\\\": \\\"nova\\\", \\\"metadata\\\": {}, \\\"status\\\": \\\"ACTIVE\\\", \\\"updated\\\": \\\"2016-08-10T18:32:23Z\\\", \\\"hostId\\\": \\\"be958de354ca4b72bb0a02694148f8d7f2d5ba965cb49e864fe63d37\\\", \\\"HUMAN_ID\\\": true, \\\"OS-SRV-USG:terminated_at\\\": null, \\\"public_v4\\\": \\\"10.8.183.233\\\", \\\"public_v6\\\": \\\"\\\", \\\"private_v4\\\": \\\"172.16.100.97\\\", \\\"interface_ip\\\": \\\"10.8.183.233\\\", \\\"name\\\": \\\"helloinstance\\\", \\\"created\\\": \\\"2016-08-10T18:32:16Z\\\", \\\"tenant_id\\\": \\\"f1dda47890754241a3e111f9b7394707\\\", \\\"region\\\": \\\"\\\", \\\"adminPass\\\": \\\"tX4SZ3gV85Sb\\\", \\\"os-extended-volumes:volumes_attached\\\": [], \\\"volumes\\\": [], \\\"config_drive\\\": \\\"\\\", \\\"human_id\\\": \\\"helloinstance\\\"}}\\n\", \"ansible_job_id\": \"501588893815.445\"}", "results_file": "/root/.ansible_async/501588893815.445", "started": 1}

NO MORE HOSTS LEFT *************************************************************
        to retry, use: --limit @pluck_os.retry

PLAY RECAP *********************************************************************
localhost                  : ok=1    changed=0    unreachable=0    failed=1 
```

**Comment by samvarankashyap** [comment]
apparently issue is with os_server module , when used with asyc gives out following extraline in async_file in .ansible_async/

```
309653937203.28231:1:No handlers could be found for logger "keystoneauth.identity.base"
```

which cannot be parsed by json.loads of async_status module . That is cause of the error .

**Comment by nitzmahone** [comment]
Yep, the control side has a filter that runs against module output to filter noise like this before the JSON load- async_wrapper needs the same filter.

**Comment by emonty** [comment]
@Shrews I think we also need to add a call in our OpenStackCloud constructor call that does all the null loggers. I agree with @nitzmahone that we should add the filter to async_wrapper - but there's no need to spam people with the keystoneauth warnings either.

**Comment by nitzmahone** [comment]
Should be fixed in devel and 2.2 RC1 by https://github.com/ansible/ansible-modules-core/pull/5107.

**Comment by smeijer** [comment]
When I use ansible from a `git checkout` I can indeed use `os_server` together with `async` & `poll: 0`, whereas `ansible 2.1.2.0` failed.

Closes some issues for me. :+1:
